components/brave_wallet/browser/eth_json_rpc_controller_unittest.cc

```diff
@@ -67,53 +68,86 @@ class EthJsonRpcControllerUnitTest : public testing::Test {
-
-  controller.SetNetwork(Network::kGoerli);
-  ASSERT_EQ(controller.GetNetwork(), Network::kGoerli);
-  ASSERT_EQ(controller.GetNetworkURL().GetOrigin(),
-            "https://goerli-infura.brave.com/");
-
-  controller.SetNetwork(Network::kKovan);
-  ASSERT_EQ(controller.GetNetwork(), Network::kKovan);
-  ASSERT_EQ(controller.GetNetworkURL().GetOrigin(),
-            "https://kovan-infura.brave.com/");
-
-  controller.SetNetwork(Network::kLocalhost);
-  ASSERT_EQ(controller.GetNetwork(), Network::kLocalhost);
-  ASSERT_EQ(controller.GetNetworkURL().GetOrigin(), "http://localhost:8545/");
+
+  controller.SetNetwork(brave_wallet::mojom::Network::Mainnet);
+  controller.GetNetwork(
+      base::BindOnce([](brave_wallet::mojom::Network network) {
+        EXPECT_EQ(network, brave_wallet::mojom::Network::Mainnet);
+      }));
+  controller.GetNetworkUrl(base::BindOnce([](const std::string& spec) {
+    EXPECT_EQ(GURL(spec).GetOrigin(), "https://mainnet-infura.brave.com/");
+  }));
+
+  controller.SetNetwork(brave_wallet::mojom::Network::Rinkeby);
+  controller.GetNetwork(
+      base::BindOnce([](brave_wallet::mojom::Network network) {
+        EXPECT_EQ(network, brave_wallet::mojom::Network::Rinkeby);
+      }));
+  controller.GetNetworkUrl(base::BindOnce([](const std::string& spec) {
+    EXPECT_EQ(GURL(spec).GetOrigin(), "https://rinkeby-infura.brave.com/");
+  }));
+
+  controller.SetNetwork(brave_wallet::mojom::Network::Ropsten);
+  controller.GetNetwork(
+      base::BindOnce([](brave_wallet::mojom::Network network) {
+        EXPECT_EQ(network, brave_wallet::mojom::Network::Ropsten);
+      }));
+  controller.GetNetworkUrl(base::BindOnce([](const std::string& spec) {
+    EXPECT_EQ(GURL(spec).GetOrigin(), "https://ropsten-infura.brave.com/");
+  }));
+
+  controller.SetNetwork(brave_wallet::mojom::Network::Goerli);
+  controller.GetNetwork(
+      base::BindOnce([](brave_wallet::mojom::Network network) {
+        EXPECT_EQ(network, brave_wallet::mojom::Network::Goerli);
+      }));
+  controller.GetNetworkUrl(base::BindOnce([](const std::string& spec) {
+    EXPECT_EQ(GURL(spec).GetOrigin(), "https://goerli-infura.brave.com/");
+  }));
+
+  controller.SetNetwork(brave_wallet::mojom::Network::Kovan);
+  controller.GetNetwork(
+      base::BindOnce([](brave_wallet::mojom::Network network) {
+        EXPECT_EQ(network, brave_wallet::mojom::Network::Kovan);
+      }));
+  controller.GetNetworkUrl(base::BindOnce([](const std::string& spec) {
+    EXPECT_EQ(GURL(spec).GetOrigin(), "https://kovan-infura.brave.com/");
+  }));
+
+  controller.SetNetwork(brave_wallet::mojom::Network::Localhost);
+  controller.GetNetwork(
+      base::BindOnce([](brave_wallet::mojom::Network network) {
+        EXPECT_EQ(network, brave_wallet::mojom::Network::Localhost);
+      }));
+  controller.GetNetworkUrl(base::BindOnce([](const std::string& spec) {
+    EXPECT_EQ(GURL(spec).GetOrigin(), "http://localhost:8545/");
+  }));
+  base::RunLoop().RunUntilIdle();
```

Comment:
there should be a test for `Custom` in here as well since that is a valid option for this method

Reply:
Since it's now `NOTREACHED()` per https://github.com/brave/brave-core/pull/9564/commits/56f6a3db242a3ce3554abfae6936fc719245a14e I think this is not needed.